config.sample.yaml

```diff
@@ -387,7 +387,9 @@ ircService:
         # through the bridge e.g. caller ID as there is no way to /ACCEPT.
         # Default: "" (no user modes)
         # userModes: "R"
-        # The format of the realname defined for users, either mxid or reverse-mxid
+        # The format of the realname defined for users.
+        # either "mxid", "reverse-mxid", or "template:TEMPLATE", where TEMPLATE must
+        # contain either of $USERID, $LOCALPART, $DISPLAY, $IRCUSER
```

Comment:
Is template limited to strictly "$USERID", "$LOCALPART", "$DISPLAY", "$IRCUSER" or can it contain prefix/suffix characters, as the help text isn't clear?

I think on the whole I would make `mxid` and `reverse-mxid` available as templating options inline with what you have there already (mxid is essentially $USERID) and then drop the whole `template:` thing entirely, so the setting looks very similar to `nickTemplate` in the end.

Reply:
Agreed; if I did this from scratch that's exactly how I would do it as well.

The reason for doing it this way is to keep backwards compatibility so users who have already configured this will keep having it working as previously. Though naturally this is at the expense of added complexity.

What do you think - keep as is or introduce a breaking change?

Reply:
We tend to favour dropping support for legacy options in the config, and adding backwards compatibility (with noisy warnings in the code instead). https://github.com/matrix-org/matrix-appservice-irc/blob/401c1ac0d308f26c7cc9fca0aa51941fbcfd16e0/src/irc/IrcServer.ts#L784-L790

At a later point, we can remove the old setting.

Reply:
Sounds reasonable, I'll take a stab at that

Reply:
How did the stabbing go?